dubbo-common/src/main/java/org/apache/dubbo/config/MetadataReportConfig.java

```diff
@@ -152,7 +153,11 @@ public URL toUrl() throws IllegalArgumentException {
         // Normalize the parameters
         map.putAll(convert(map, null));
         // put the protocol of URL as the "metadata"
-        map.put("metadata", url.getProtocol());
+        String metadataProtocol = isEmpty(url.getProtocol()) ? map.get(PROTOCOL_KEY) : url.getProtocol();
+        if (isEmpty(metadataProtocol)) {
+            throw new IllegalArgumentException("Please specify valid protocol for metadata report." + address);
+        }
```

Comment:
It is recommended to be better as a config validation, and complete the logic in ConfigValidationUtils#validateMetadataConfig()